x-pack/libbeat/tests/integration/collector_test.go

```diff
@@ -0,0 +1,177 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package integration
+
+import (
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/exporter/debugexporter"
+	"go.opentelemetry.io/collector/otelcol"
+
+	"github.com/elastic/beats/v7/libbeat/otelbeat/beatconverter"
+	"github.com/elastic/beats/v7/libbeat/otelbeat/providers/fbprovider"
+	"github.com/elastic/beats/v7/libbeat/tests/integration"
+	"github.com/elastic/beats/v7/x-pack/filebeat/fbreceiver"
+)
+
+var schemeMap = map[string]string{
+	"filebeat": "fb",
+}
+
+type TestCollector struct {
+	t          *testing.T
+	tempDir    string
+	otelcol    *otelcol.Collector
+	wg         sync.WaitGroup
+	configFile string
+	beatname   string
+}
+
+// NewTestCollector configures and returns an otel collector intended for testing only
+// It accepts beatname and configuration
+func NewTestCollector(t *testing.T, beatname string, config string) (*TestCollector, error) {
+
+	// create a temp dir
+	tempDir := integration.CreateTempDir(t)
+	// stdoutFile, err := os.Create(filepath.Join(tempDir, "stdout"))
+	// require.NoError(t, err, "error creating stdout file")
+	// stderrFile, err := os.Create(filepath.Join(tempDir, "stderr"))
+	// require.NoError(t, err, "error creating stderr file")
+
+	// create a config file
+	configFile := filepath.Join(tempDir, beatname+".yml")
+	// write configuration to a file
+	if err := os.WriteFile(configFile, []byte(config), 0o644); err != nil {
+		t.Fatalf("cannot create config file '%s': %s", configFile, err)
+	}
+	// adds scheme name as prefix to the configfile
+	beatCfg := schemeMap[beatname] + ":" + configFile
+	// get collector settings
+	set := getCollectorSettings(beatCfg)
+	// get new collector instance
+	otelcol, err := otelcol.NewCollector(set)
```

Comment:
These bits seems to be the same as ReloadConfig, maybe we can just call the function here instead?